codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
It is better to raise exception when both basePackage and folder is empty. 
he need to provide at least one [basePackage or folder ].

with the current change if the user purposefully made basePackage empty and given detailed folder structure, here we are appending "com.paypal.selion.testcomponents" even though he provided folder structure.